web/src/title/Loading.svelte

```diff
@@ -0,0 +1,49 @@
+<script lang="ts">
+  // TODO Upstream to svelte-utils
+
+  // Block the rest of the page while 'loading' is non-empty
+
+  export let loading: string[];
+
+  let inner: HTMLDivElement;
+
+  $: if (loading.length > 0 && inner) {
+    inner.scrollTop = inner.scrollHeight;
+  }
+</script>
+
+{#if loading.length > 0}
+  <div class="outer">
+    <div class="inner" bind:this={inner}>
+      {#each loading as msg}
+        <p>{msg}</p>
```

Comment:
The styling is pretty bad. The messages have whitespace indenting at the beginning to indicate nestedness. Probably should return little JSON blobs like ` { nesting: 2, msg: "foo" }` or an actual tree structure, and maybe even glue to some existing JS flamegraph library